### ISSUE TITLE
Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work.

### ISSUE THREAD

**Issue by piotrkochan** [mattn/go-sqlite3#855]
go-sqlite3 - tag v1.14.3

I'm building app in the goreleaser in docker environment. I read on README:

```
If you want to build your app using go-sqlite3, you need gcc. However, after you have built and installed go-sqlite3 with go install github.com/mattn/go-sqlite3 (which requires gcc), you can build your app without relying on gcc in future.
```

I'm running:

```
CGO_ENABLED=1 go install github.com/mattn/go-sqlite3
goreleaser --snapshot --rm-dist
```

goreleaser.yml:

```
builds:
  binary: [...]
  env:
  - CGO_ENABLED=0
  goarch:
    - amd64
  goos:
  - linux
```

and the error is:

```
./myapp
2020-09-17T14:24:42.648+0200    INFO    app/db.go:15    using database ./app.sqlite
panic: Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub
```

**Comment by piotrkochan** [comment]
```
CGO_ENABLED=1 go install github.com/mattn/go-sqlite3
go build
./app
```

gives same error so it is not related to the goreleaser itself

**Comment by rittneje** [comment]
I don't think the statement in the goreleaser README is true. (Or at least, isn't true anymore.)

```
CGO_ENABLED=1 go install github.com/mattn/go-sqlite3
go build
./app
```

In this example, when you run `go build` (presumably with CGO_ENABLED=0), the Go compiler recompiles this library _without_ cgo. To my knowledge, there is no way to force the Go compiler to use a pre-built version of a library, and if detects any changes (including enabling/disabling cgo), it will recompile.

**Comment by iamstefin** [comment]
I'm facing the same issue.

**Comment by ThrillQuks** [comment]
same issues here

**Comment by Clivern** [comment]
same issues here!

**Comment by bindian0509** [comment]
@aviddiviner Thanks man my tests were not building I was using sqlite database for go application, getting this error - 

`failed to initialize database, got error %v, /var/lib/jenkins/workspace/GoBuildTest/repository/repository.go:80%!(EXTRA *errors.errorString=Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub)`

I was using it as a part of jenkins build process, and my jenkins server was running on EC2 ubuntu machine, with **build-essential** this worked fine.

**Comment by Hirosaji** [comment]
If you are using M1 Mac and have specified the environment variable GOARCH=amd64, you will need to change it as follows

`go env -w GOARCH=arm64`

**Comment by matiasdev30** [comment]
same issues here

**Comment by ayoubzulfiqar** [comment]
> Has anyone managed to solve this in a windows environment?

```go env -w CGO_ENABLED=1```

**Comment by margani** [comment]
I has this issue in Windows and did the following to fix it:

tl;dr: 
1. Run `go env -w CGO_ENABLED=1`
2. Download and install [tdm64-gcc-5.1.0-2.exe](https://sourceforge.net/projects/tdm-gcc/files/TDM-GCC%20Installer/tdm64-gcc-5.1.0-2.exe/download) with default options (to add to PATH env variable), and restart your terminal or IDE session.

Longer explanation:

* Thanks to @ayoubzulfiqar, I ran `go env -w CGO_ENABLED=1`
* Then when I run `go run .`, it shows the error: `cgo: C compiler "gcc" not found: exec: "gcc": executable file not found in %PATH%`
* I search the web, and found [an article](https://dev.to/gamegods3/how-to-install-gcc-in-windows-10-the-easier-way-422j) that I can use [MinGW (`mingw-get-setup.exe`)](https://sourceforge.net/projects/mingw/files/Installer/), and I followed the article, and I had GCC available, but then `go run .` didn't work, it showed the error: `cc1.exe: sorry, unimplemented: 64-bit mode not compiled in`
* So I searched the error, and found this [SO answer](https://stackoverflow.com/a/39867163/87514) that said TDM-GCC only works in Windows x64, so I downloaded [tdm64-gcc-5.1.0-2.exe](https://sourceforge.net/projects/tdm-gcc/files/TDM-GCC%20Installer/tdm64-gcc-5.1.0-2.exe/download) and installed it, and removed MinGW from my System PATH environment variable, and now I can build and run my go app using go-sqlite3 package.

**Comment by maximka200** [comment]
> I has this issue in Windows and did the following to fix it:
> 
> tl;dr:
> 
> 1. Run `go env -w CGO_ENABLED=1`
> 2. Download and install [tdm64-gcc-5.1.0-2.exe](https://sourceforge.net/projects/tdm-gcc/files/TDM-GCC%20Installer/tdm64-gcc-5.1.0-2.exe/download) with default options (to add to PATH env variable), and restart your terminal or IDE session.
> 
> Longer explanation:
> 
> * Thanks to @ayoubzulfiqar, I ran `go env -w CGO_ENABLED=1`
> * Then when I run `go run .`, it shows the error: `cgo: C compiler "gcc" not found: exec: "gcc": executable file not found in %PATH%`
> * I search the web, and found [an article](https://dev.to/gamegods3/how-to-install-gcc-in-windows-10-the-easier-way-422j) that I can use [MinGW (`mingw-get-setup.exe`)](https://sourceforge.net/projects/mingw/files/Installer/), and I followed the article, and I had GCC available, but then `go run .` didn't work, it showed the error: `cc1.exe: sorry, unimplemented: 64-bit mode not compiled in`
> * So I searched the error, and found this [SO answer](https://stackoverflow.com/a/39867163/87514) that said TDM-GCC only works in Windows x64, so I downloaded [tdm64-gcc-5.1.0-2.exe](https://sourceforge.net/projects/tdm-gcc/files/TDM-GCC%20Installer/tdm64-gcc-5.1.0-2.exe/download) and installed it, and removed MinGW from my System PATH environment variable, and now I can build and run my go app using go-sqlite3 package.

it works, thx

**Comment by Kaloszer** [comment]
@margani works! Thanks

Note that the installer tells you to get the new ver from:
https://jmeubank.github.io/tdm-gcc/download/

**Comment by michal-kapala** [comment]
`gorm.io/driver/sqlite` still depends on this

**Comment by nevenag** [comment]
Is there a workaround for docker on Mac M1 with amd64?

**Comment by prof79** [comment]
Heroes of the day!!! 🎉 @margani @Kaloszer @ayoubzulfiqar @jmeubank
Rarely seen guidance so to-the-point and excellent installer of the tdm-gcc team, instantly up and running
Go really has a slick (and sick ^^) ORM with GORM!

**Comment by samjewell** [comment]
@rittneje said [here](https://github.com/mattn/go-sqlite3/issues/855#issuecomment-694270384):

> I don't think the statement in the README is true. (Or at least, isn't true anymore.)

Seems that line was added to the readme in https://github.com/mattn/go-sqlite3/pull/250, back in 2015. Not much extra context there, but quickly searching online I agree with your conclusion.
